di_website/events/models.py

```diff
@@ -44,22 +48,41 @@ class Meta:
         verbose_name = "Event"
         verbose_name_plural = "Events"
 
-
 """ List of all events that have been created from events page """
+class EventIndexPage(StandardPage):    
+    body = StreamField(BaseStreamBlock(), verbose_name="Page Body", blank=True)    
 
-
-class EventIndexPage(StandardPage):
-    
-    template = "events/events_index_page.html"
-
-    subpage_types = ['EventPage']
+    content_panels = StandardPage.content_panels + [
+        StreamFieldPanel('body'),        
+    ]
 
     def get_context(self,request):
-        context = super().get_context(request)
-        context['events'] = EventPage.objects.live().all()
-        context['active_page'] = self.title
+        context = super(EventIndexPage, self).get_context(request)
+        page = request.GET.get('page', None)
+        topic_filter = request.GET.get('topic', None)
+        if topic_filter:
+            events = EventPage.objects.live().filter(topics__slug=topic_filter)
+        else:
+            events = EventPage.objects.live()
+
+        paginator = Paginator(events, 10)
+        try:
+            context['events'] = paginator.page(page)
+        except PageNotAnInteger:
+            context['events'] = paginator.page(1)
+        except EmptyPage:
+            context['events'] = paginator.page(paginator.num_pages)
+
+        blog_content_type = ContentType.objects.get_for_model(EventPage)
```

Comment:
I removed this variable because it feeds into tags which are not on the event listing template http://development-initiatives.surge.sh/page-templates/11b-events-listing